components/shared/ui/menu.tsx

```diff
@@ -1,53 +1,99 @@
 import Link from "next/link";
 import { FC, useState } from "react";
-import { mainColorBgClass } from "../../../lib/constants/colors";
+import { mainColorBgClass, mainColorBorderClass } from "../../../lib/constants/colors";
 import { useSiteCodename } from "../siteCodenameContext";
 import { Block_Navigation } from "../../../models";
 import { createItemSmartLink } from "../../../lib/utils/smartLinkUtils";
 import { ChevronDownIcon, Bars3Icon } from "@heroicons/react/24/solid";
+import { useRouter } from "next/router";
 
 type Link = Readonly<Block_Navigation>;
 
 type Props = Readonly<{
   item: Link;
 }>
 
+type MenuItemsProps = Readonly<{
+  items: Block_Navigation[];
+  activeMenu: string | number;
+  handleClick: (menuId: string | number) => void;
+}>
+
 type DropdownMenuProps = Readonly<{
   links: ReadonlyArray<Link>;
 }>;
 
-const DropdownMenu: FC<DropdownMenuProps> = props => {
+const MenuItems: FC<MenuItemsProps> = props => {
+  const siteCodename = useSiteCodename();
+  const router = useRouter();
+
   return (
     <>
-      {props.links.map(link => (
-        <Link
-          key={link.system.codename}
-          {...link.elements.openInANewWindow.value[0] ? { rel: "noopener noreferrer", target: "_blank" } : {}}
-          href={link.elements.externalLink.value ? link.elements.externalLink.value : "/" + link.elements.pageLink.linkedItems[0].elements.url.value}
-          className="block p-3 rounded-lg hover:bg-gray-50"
-        >
-          <div className="font-semibold">{link.elements.label.value}</div>
-          <span className="text-sm text-gray-500">{link.elements.caption.value}</span>
-        </Link>
+      {props.items.map((link, i) => (
+        <li key={i} className={`${"/" + link.elements.pageLink.linkedItems[0]?.elements.url.value === router.asPath ? "" : "border-l-transparent border-t-transparent"}
+        border-gray-500 border-l-8 border-t-0 md:border-t-8 md:border-l-0 h-full ${mainColorBgClass[siteCodename]} group grow`} onClick={() => props.handleClick(i)}>
```

Comment:
you can use `${mainColorBorderClass[siteCodename]}` instead of `border-gray-500`. As I don't find gray color so exciting

however you need to adjust values in `mainColorBorderClass` as  there is also `border-green-300` which is probably the same color as background and therefore not visible

`Border-green-400` works for my eyes, did not test other values :) 

Or you can create new colors class for this border :) What do you say?

Reply:
We might define the grey color, but we are using "border-", "border-l*", "border-t-*" + also "hover:border*" and tailwind is now set to concat the string in order to create class name - that is why I decided to use dark gres - it firs all defined colors currently and that is why it is not deported in the constant, it is specific to menu itself.